File: 2418-sort-the-people/2418-sort-the-people.swift
class Solution {
	func sortPeople(_ names: [String], _ heights: [Int]) -> [String] {
		var map: [(String, Int)] = []
		for i in 0 ..< names.count {
			map.append((names[i], heights[i]))
		}

		map.sort { pair1, pair2 in pair1.1 > pair2.1 }
		let result = map.map { pair in
			pair.0
		}
//		print(map)
//		print(result)
		return result
	}
}

File: 2418-sort-the-people/2418-sort-the-people.java
class Pair implements Comparable<Pair> {
    String word;
    Integer number;

    public Pair(String _word, Integer _number) {
        this.word = _word;
        this.number = _number;
    }


    @Override
    public int compareTo(Pair o) {
        return number.compareTo(o.number);
    }
}

class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        List<Pair> pairs = new ArrayList<>();
        for (int i = 0; i < names.length; i++) {
            pairs.add(new Pair(names[i], heights[i]));
        }
        pairs.sort(Comparator.reverseOrder());
        String[] result = new String[pairs.size()];
        for (int i = 0; i < pairs.size(); i++) {
            result[i] = pairs.get(i).word;
        }
        return result;
    }
}